packages/components/src/dropdown/index.tsx

```diff
@@ -123,34 +167,51 @@ const UnconnectedDropdown = (
 			tabIndex={ -1 }
 			style={ style }
 		>
-			{ renderToggle( args ) }
-			{ isOpen && (
-				<Popover
-					position={ position }
-					onClose={ close }
-					onFocusOutside={ closeIfFocusOutside }
-					expandOnMobile={ expandOnMobile }
-					headerTitle={ headerTitle }
-					focusOnMount={ focusOnMount }
-					// This value is used to ensure that the dropdowns
-					// align with the editor header by default.
-					offset={ 13 }
-					anchor={
-						! popoverPropsHaveAnchor
-							? fallbackPopoverAnchor
-							: undefined
-					}
-					variant={ variant }
-					{ ...popoverProps }
-					className={ classnames(
-						'components-dropdown__content',
-						popoverProps?.className,
-						contentClassName
-					) }
-				>
-					{ renderContent( args ) }
-				</Popover>
-			) }
+			{ containerRef.current && isOpen && showBackdrop
+				? /* The backdrop is rendered via portal because otherwise its fixed
+				   * position would be often affected by parent containers acting as
+				   * containing blocks.
+				   * See https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block
+				   */
+				  createPortal(
+						<DropdownPointerEventsCapture
+							onClick={ () => setShowBackdrop( false ) }
```

Comment:
The backdrop should always disappear when the dragging ends, but if it was to stick around for any reason, the first click would dismiss it.

This is mostly to make sure that the page doesn't end up in a state in which most of the UI isn't clickable